File: AED-2/Arvores/bst/TAD-flexivel/matriz-flexivel/java/diagonais.java
package matriz.java;

public class diagonais {
    void diagonalPrincipal() {
        Celula temp = inicio;
        for (int i = 0; i < linhas && i < colunas; i++) {
            Celula p = temp;
            for (int j = 0; j < i; j++) p = p.dir;
            System.out.print(p.elemento + " ");
            temp = temp.inf;
        }
        System.out.println();
    }

    void diagonalSecundaria() {
        Celula temp = inicio;
        for (int i = 0; i < linhas && i < colunas; i++) {
            Celula p = temp;
            for (int j = 0; j < colunas - i - 1; j++) p = p.dir;
            System.out.print(p.elemento + " ");
            temp = temp.inf;
        }
        System.out.println();
    }

}

File: AED-2/Arvores/bst/TAD-flexivel/matriz-flexivel/c/diagonais.c
void mostrarDiagonalPrincipal(Matriz* mat) {
    Celula *temp = mat->inicio;
    for (int i = 0; i < mat->linhas && i < mat->colunas; i++) {
        Celula *p = temp;
        for (int j = 0; j < i; j++) p = p->dir;
        printf("%d ", p->elemento);
        temp = temp->inf;
    }
    printf("\n");
}

void mostrarDiagonalSecundaria(Matriz* mat) {
    Celula *temp = mat->inicio;
    for (int i = 0; i < mat->linhas && i < mat->colunas; i++) {
        Celula *p = temp;
        for (int j = 0; j < mat->colunas - i - 1; j++) p = p->dir;
        printf("%d ", p->elemento);
        temp = temp->inf;
    }
    printf("\n");
}
